jung-visualization/src/main/java/edu/uci/ics/jung/visualization/subLayout/TreeCollapser.java

```diff
@@ -49,6 +52,29 @@ public static void expand(MutableCTreeNetwork tree, CTreeNetwork subTree) {
             ? tree.inEdges(subTree).iterator().next() // THERE CAN BE ONLY ONE
             : null;
     tree.removeNode(subTree);
-    TreeUtils.addSubTree(tree, subTree, parent.orElse(null), parentEdge);
+    if (parent.isPresent()) {
+      TreeUtils.addSubTree(tree, subTree, parent.get(), parentEdge);
+    } else {
+      // tree is empty, so just copy all of subTree into tree
+      copyInto(tree, subTree);
+    }
+  }
+
+  private static <N, E> void copyInto(MutableCTreeNetwork<N, E> to, CTreeNetwork<N, E> from) {
+    from.root()
+        .ifPresent(
+            root -> {
+              to.addNode(root);
+              // Connect each successively deeper node to its parent
+              Iterable<N> insertionOrder =
+                  TreeTraverser.using(from::successors)
```

Comment:
TreeTraverser is deprecated and will be replaced with common.graph.Traverser.forTree(); please use this instead.  (You can just pass in 'from' instead of needing to use the method reference.)

Reply:
Sadly, JUNG currently depends on Guava 22.0, which doesn't have `Traverser`.

My gut feeling is to add a TODO here to replace this `TreeTraverser` usage with `Traverser` when the Guava version gets bumped. WDYT?

Reply:
Oh.  Then we should bump the Guava version.  There's nothing sacred about 22.0.  We can either bump it now in a separate PR and then merge that change into this version, or you can add a TODO and then we can change this again.

Also note that if we take my suggestion above (of returning the expanded graph/modifying what ) this all becomes moot anyway, because the need for copyInto() arguably disappears.  :)

Reply:
I went and added the TODO before I saw this comment, but I think the idea of bumping the Guava version in a separate PR and then merging/rebasing sounds good! I'll raise such a PR in a bit.